### ISSUE TITLE
Volume is stored as `int` in the catalog, while it should be a `float`

### ISSUE THREAD

**Issue by yohplala** [nautechsystems/nautilus_trader#502]
### Expected Behavior
After a 1st contact with Nautilus trader to load Binance data, I checked stored data, and I can see that the column 'volume' only contains `int` while it shoud contain `float`.

### Actual Behavior
Well, it contains `int`.

### Steps to Reproduce the Problem
I could see that on example provided in #500.
Here is code. Data to use can be that provided in PR #501 (I enclosed 2 files from Binance official repo).
```python
import os, shutil
import pandas as pd

from nautilus_trader.persistence.catalog import DataCatalog
from nautilus_trader.persistence.external.core import process_files
from nautilus_trader.persistence.external.readers import CSVReader
from nautilus_trader.backtest.data.wranglers import BarDataWrangler
from nautilus_trader.model.enums import BarAggregation, PriceType
#from tests.test_kit.stubs import TestStubs
from nautilus_trader.backtest.data.providers import TestInstrumentProvider
from nautilus_trader.model.identifiers import Symbol, Venue
from nautilus_trader.model.identifiers import InstrumentId
from nautilus_trader.model.data.bar import BarSpecification, BarType

# https://docs.nautilustrader.io/user-guide/backtesting
DATA_DIR = "~/Documents/code/data/binance/data/spot/monthly/klines/ADABTC/1m"
CATALOG_PATH = os.path.expanduser("~/Documents/code/data/nautilus/catalog")

# Clear if it already exists, then create fresh
if os.path.exists(CATALOG_PATH):
    shutil.rmtree(CATALOG_PATH)
os.makedirs(CATALOG_PATH)
catalog = DataCatalog(CATALOG_PATH)

adabtc_id = InstrumentId(Symbol("ADA/BTC"), Venue("BINANCE"))
bar_spec_1min_last = BarSpecification(1, BarAggregation.MINUTE, PriceType.LAST)
bar_type = BarType(adabtc_id, bar_spec_1min_last)
#bar_type = TestStubs.bartype_adabtc_binance_1min_last()
instrument = TestInstrumentProvider.adabtc_binance()
wrangler = BarDataWrangler(bar_type, instrument)
def binance_csvzip_parser2(data):
    data['timestamp'] = data['timestamp'].astype('datetime64[ms]')
#    data = data[['timestamp', 'open', 'high', 'low', 'close', 'volume']]
    bars = wrangler.process(data.set_index("timestamp"))
    return bars

binance_spot_header = ['timestamp','open','high','low','close', 'volume',
                       'ts_close', 'quote_volume', 'n_trades',
                       'taker_buy_base_volume', 'taker_buy_quote_volume',
                       'ignore']

binance_spot_csv_reader = CSVReader(block_parser=binance_csvzip_parser2,
                                    header=binance_spot_header)

process_files(
    glob_path=f"{DATA_DIR}/*.zip",
    reader=binance_spot_csv_reader,
    catalog=catalog,
)
```
### Specifications

- OS Platform: Ubuntu 20.04
- NautilusTrader Version: 0.134.0

**Comment by yohplala** [comment]
Hi,
Sorry for above incomplete message, it started to be a bit late yesterday evening.
So actually, I had already prepared a 2nd commit at the time of writing, with an additional test case to validate `BarDataWrangler` with binance data.
This is with this test I could see the trouble.

Please, find this test case in this PR from my remote branch 'CSV_wrangler_reviewed' to my other remote branch 'binance_csv', name `class TestBarDataWranglerHeaderless:` in `tests/unit_tests/backtest/test_backtest_data_wranglers.py`
([direct permalink](https://github.com/yohplala/nautilus_trader-1/blob/d15151b8e1a3dce28a76e046f14716c54512baa6/tests/unit_tests/backtest/test_backtest_data_wranglers.py#L221))

When pytesting it, I get following error.
```bash
poetry run pytest tests/unit_tests/backtest/test_backtest_data_wranglers.py 

>       assert bars[0].volume == Quantity.from_str("36304.2")
E       AssertionError: assert Quantity('36304') == Quantity('36304.2')
E        +  where Quantity('36304') = Bar(ADA/BTC.BINANCE-1-MINUTE-LAST-EXTERNAL,0.00002853,0.00002854,0.00002851,0.00002854,36304,1637971200000000000).volume
E        +  and   Quantity('36304.2') = <cyfunction Quantity.from_str at 0x7fcf097ff790>('36304.2')
E        +    where <cyfunction Quantity.from_str at 0x7fcf097ff790> = Quantity.from_str
```

**Comment by yohplala** [comment]
I ought not to have time today for working on it. If you don't either, I should be able to investigate further tomorrow evening again. (my 1st guess is that probably `dtype` are specified for each column when recording parquet file, and that `int` is set for 'volume' column? But I don't know yet in which file this is specified to check, and today I will be short on time :))

**Comment by yohplala** [comment]
(I couldn't help deep diving a bit more)

Ok, I found it (I think so, I am not on my machine to test, only browsing the files through a web browser)
So, in `BarDataWranglers`,
 - we have to provide an `Instrument` used in its `__init__` as can be seen [here](https://github.com/nautechsystems/nautilus_trader/blob/2d602dd98fff248ba524bb28a38aebcb1de384be/nautilus_trader/backtest/data/wranglers.pyx#L314)
 - this `Instrument` specifies the precision for `Quantity` as can be seen [here](https://github.com/nautechsystems/nautilus_trader/blob/2d602dd98fff248ba524bb28a38aebcb1de384be/nautilus_trader/backtest/data/wranglers.pyx#L395).
   `volume=Quantity(values[4], self.instrument.size_precision)`

In my test cases (the one in PR #501 and the one in the PR not yet pushed), I am using the pre-defined [`adabtc_binance()`](https://github.com/nautechsystems/nautilus_trader/blob/2d602dd98fff248ba524bb28a38aebcb1de384be/nautilus_trader/backtest/data/providers.py#L60) `Intrument` from [`providers.py`](https://github.com/nautechsystems/nautilus_trader/blob/master/nautilus_trader/backtest/data/providers.py)

Guess what? Well obviously, [line 78](https://github.com/nautechsystems/nautilus_trader/blob/2d602dd98fff248ba524bb28a38aebcb1de384be/nautilus_trader/backtest/data/providers.py#L78):
`size_precision=0,`

Question is now: in your opinion, what is the best practise?
 - should I set a pre-defined precision?
 - or can I let Nautilus decides on its own what should be the precision?
I can see a `Quantity` is able to decide on its own but it is not called this way in `BarDataWrangler`.

Thanks in advance for your feedback.
Bests,

**Comment by yohplala** [comment]
Ok, being rational here I think, I changed precision of `adabtc_binance()` instrument to 1e-8, that of the binance data file, as can be seen on extract below.
```python
#   timestamp,      open,      high,       low,     close,        volume, 
1637971200000,0.00002853,0.00002854,0.00002851,0.00002854,36304.20000000,1637971259999,1.03547608,79,19523.00000000,0.55694227,0
```
Fixed by #503 
Bests

**Comment by cjdsellers** [comment]
Hi @yohplala 

I think its best practice to set the correct precision for ADABTC on the stub test instrument. We could create a flow where a `.from_str` method is called, however I don't think we can rely on the veracity of the datas precision in all cases, so best we define the precision in one place (on the instrument).

**Comment by yohplala** [comment]
> I think its best practice to set the correct precision for ADABTC on the stub test instrument.

I think it is what I did in #503, changing the precision in `adabtc_binance()` from `data/providers.py`. Please, confirm me if this is ok.

**Comment by cjdsellers** [comment]
Yes, this is fine. I think the `.zip` files in that diff need changing to plain CSV then it should be good to merge.

**Comment by cjdsellers** [comment]
Fixed from #503 

Many thanks!
